cpp/src/io/parquet/page_enc.cu

```diff
@@ -933,6 +940,28 @@ static __device__ void PlainBoolEncode(page_enc_state_s *s,
   }
 }
 
+/**
+ * @brief Converts a sys_time<nanoseconds> into a pair with nanoseconds since midnight and number of
+ * Julian days. Does not deal with time zones. Used by INT96 code.
+ *
+ * @param ns number of nanoseconds since epoch
+ * @return std::pair<nanoseconds,days> where nanoseconds is the number of nanoseconds
+ * elapsed in the day and days is the number of days from Julian epoch.
+ */
+static __device__ std::pair<simt::std::chrono::nanoseconds, simt::std::chrono::days>
+convert_nanoseconds(simt::std::chrono::sys_time<simt::std::chrono::nanoseconds> const ns)
+{
+  constexpr int64_t JulianEpochOffsetDays = INT64_C(2440588);
```

Comment:
It might be a little less magic to do something like this:
```
constexpr
auto
jdiff()
{
    using namespace date;
    using namespace std::chrono_literals;
    return sys_days{jan/1/1970} - (sys_days{nov/24/-4713} + 12h);
}
```

From: https://stackoverflow.com/a/33964462/11341974